app/src/main/java/br/org/cesar/discordtime/stickysessions/domain/interactor/AddNote.java

```diff
@@ -0,0 +1,19 @@
+package br.org.cesar.discordtime.stickysessions.domain.interactor;
+
+import br.org.cesar.discordtime.stickysessions.domain.model.Note;
+import br.org.cesar.discordtime.stickysessions.domain.repository.NoteRepository;
+import io.reactivex.Single;
+
+public class AddNote extends UseCase<Note, Note> {
+
+    private NoteRepository mRepository;
+
+    public AddNote(NoteRepository repository) {
+        mRepository = repository;
+    }
+
+    @Override
+    public Single<Note> execute(Note note) {
+        return mRepository.addNote(note);
```

Comment:
Maybe we should check if the sessionId that this note carries is a valid one? Or if the topic is a valid one?

Reply:
I think it would be enough for now to pass current Session instance to the method and then check those two things. In future, we could check that direclty on SessionRepository cache.

Reply:
Please also include tests to this UseCase and make sure we're checking that both `sessionId` and `topic` match with current session info. An `Single.error` should be returned otherwise.